services/datamanager/builtin/builtin.go

```diff
@@ -539,30 +544,33 @@ func (svc *builtIn) uploadData(cancelCtx context.Context, intervalMins float64)
 			case <-svc.syncTicker.C:
 				svc.lock.Lock()
 				if svc.syncer != nil {
-					svc.sync(cancelCtx)
+					// If selective sync is not enabled (false), we default to not sync until the trigger
+					// condition has been checked. Otherwise, the default behavior is to sync.
+					shouldSync := !svc.selectiveSyncEnabled
+					// If selective sync is enabled and the sensor has been properly initialized,
+					// try to get the reading from the selective sensor that indicates whether to sync
+					if svc.syncSensor != nil && svc.selectiveSyncEnabled {
+						shouldSync = readyToSync(cancelCtx, svc.syncSensor, svc.logger)
+					}
+					if shouldSync {
+						svc.sync()
+					}
 				}
 				svc.lock.Unlock()
 			}
 		}
 	})
 }
 
-func (svc *builtIn) sync(ctx context.Context) {
+func (svc *builtIn) sync() {
 	svc.flushCollectors()
-	shouldSync := true
-	// If selective sync is enabled and the sensor has been properly initialized,
-	// try to get the reading from the selective sensor that indicates whether to sync
-	if svc.syncSensor != nil {
```

Comment:
Since we're moving this out of `sync`, then a manual Sync() API call will perform a sync no matter what. I think this makes sense, but should note that that'll cause unexpected behavior if automated sync is also enabled (which was true before the selective syncer was introduced)

Reply:
Makes sense, added comment